### PR TITLE
Fully qualify type references; remove usings

### PR DESCRIPTION
**Pull request by simonmckenzie**
This updates the interface generation to always fully qualify type references, and no longer generate "using" statements in generated interfaces.

This makes the interface generation resilient to namespace collisions and other namespace edge-cases (like resolving global usings in parent assemblies). It also means the generator is now able to correctly resolve types with duplicate names

Addresses #55

### CODE REVIEW

File: AutomaticInterface/Tests/GeneratorTests.cs [hunk 1]
```diff
@@ -1260,16 +1091,13 @@ public string CMethod<T, T1, T2, T3, T4>(string x, string y)
...
-
             namespace AutomaticInterfaceExample
             {
-                [GeneratedCode("AutomaticInterface", "")]
+                [global::System.CodeDom.Compiler.GeneratedCode("AutomaticInterface", "")]
                 public partial interface IDemoClass
                 {
                     /// <inheritdoc />
-                    string CMethod<T, T1, T2, T3, T4>(string x, string y) where T : class where T1 : struct where T3 : DemoClass where T4 : IDemoClass;
+                    string CMethod<T, T1, T2, T3, T4>(string x, string y) where T : class where T1 : struct where T3 : global::AutomaticInterfaceExample.DemoClass where T4 : IDemoClass;
```
Comment (simonmckenzie) [inline AutomaticInterface/Tests/GeneratorTests.cs:1100]: This is the one place where types aren't fully qualified - `IDemoClass` doesn't exist at the time of generation, so it can't be resolved. This is OK though, since the reference is within the scope of `IDemoClass`, so no collision is possible

File: AutomaticInterface/AutomaticInterface/Builder.cs [hunk 2]
```diff
@@ -18,6 +18,15 @@ public static class Builder
                 | SymbolDisplayParameterOptions.IncludeParamsRefOut
         );
 
+    private static readonly SymbolDisplayFormat TypeDisplayFormat =
```
Comment (simonmckenzie) [inline AutomaticInterface/AutomaticInterface/Builder.cs:21]: New display format for types. Based on `SymbolDisplayFormat.FullyQualifiedFormat`, but with support for nullables, and without unnecessary non-type parameters

File: AutomaticInterface/AutomaticInterface/InterfaceBuilder.cs [hunk 3]
```diff
@@ -103,21 +94,16 @@ public string Build()
...
             cb.AppendLine($"namespace {nameSpaceName}");
             cb.AppendLine("{");
 
             cb.Indent();
 
             cb.AppendAndNormalizeMultipleLines(classDocumentation);
 
-            cb.AppendLine($"[GeneratedCode(\"AutomaticInterface\", \"\")]");
+            cb.AppendLine(
+                "[global::System.CodeDom.Compiler.GeneratedCode(\"AutomaticInterface\", \"\")]"
```
Comment (simonmckenzie) [inline AutomaticInterface/AutomaticInterface/InterfaceBuilder.cs:105]: This was a choice - it seemed simpler and more concise to fully qualify this too, since otherwise there's only be one using in the interface file, `using System.CodeDom.Compiler;`

File: AutomaticInterface/Tests/GeneratorTests.cs [hunk 4]
```diff
@@ -257,61 +245,12 @@ public partial interface IDemoClass
         GenerateCode(code).Should().Be(expected);
     }
 
-    [Fact]
-    public void AddsUsingsToInterface()
```
Comment (simonmckenzie) [inline AutomaticInterface/Tests/GeneratorTests.cs:261]: Obsolete test

File: AutomaticInterface/Tests/GeneratorTests.cs [hunk 5]
```diff
@@ -1171,52 +1047,7 @@ public partial interface IDemoClass
...
-                    
-                }
-            }
-
-            """;
-        GenerateCode(code).Should().Be(expected);
-    }
-
-    [Fact]
-    public void WorksWithFileScopedNamespace()
```
Comment (simonmckenzie) [inline AutomaticInterface/Tests/GeneratorTests.cs:1184]: Moved into `GeneratorTests.TypeResolution`

File: AutomaticInterface/Tests/GeneratorTests.cs [hunk 6]
```diff
@@ -34,13 +34,6 @@ private static string GenerateCode(string code)
             .Should()
             .BeEmpty();
 
-        sourceDiagnostics
-            .Where(d => d.Severity == DiagnosticSeverity.Error)
-            .Where(x => x.Id != "CS0246")
-            .Where(x => !MissingUsingsAreOk(x))
-            .Should()
-            .BeEmpty();
```
Comment (simonmckenzie) [inline AutomaticInterface/Tests/GeneratorTests.cs:42]: This check was already covered in the broader check on lines 31-35:

https://github.com/codecentric/net_automatic_interface/blob/e151b0564933ca97e5685f4d93648adab5ce1ccb/AutomaticInterface/Tests/GeneratorTests.cs#L31-L35

**Comment by ChristianSauer** [comment]
Looks very good, thank you so much!
